src/Microsoft.AspNetCore.ResponseCaching/ResponseOutput/WriteOnlyShardStream.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.ResponseCaching.Internal
+{
+    internal class WriteOnlyShardStream : Stream
+    {
+        private readonly List<byte[]> _shards = new List<byte[]>();
+        private readonly MemoryStream _bufferStream = new MemoryStream();
+        private readonly int _shardSize;
```

Comment:
Going to set a static limit of 84k right now. Can make this configurable in the future if the need arises.